### PR TITLE
Swift 3.0 support

### PR DESCRIPTION
**Pull request by PChmiel**
Updated to Swift 3.0, added tests for API methods

### CODE REVIEW

File: Snappy/ImagesViewController.swift [hunk 1]
```diff
@@ -35,21 +35,21 @@ class ImagesViewController: UITableViewController {
     
     func pullToRefreshSetup() {
         refreshControl = UIRefreshControl()
-        refreshControl?.backgroundColor = UIColor.whiteColor()
-        refreshControl?.tintColor = UIColor.blackColor()
+        refreshControl?.backgroundColor = UIColor.white
```
Comment (sunshinejr) [inline Snappy/ImagesViewController.swift]: here we can even skip the `UIColor`, with just `.white` (below as well)

File: Snappy/ImagesViewController.swift [hunk 2]
```diff
@@ -73,15 +73,15 @@ class ImagesViewController: UITableViewController {
         }
     }
  
-    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCellWithIdentifier(imageCellIdentifier, forIndexPath: indexPath)
-        cell.textLabel?.text = images[indexPath.row]
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: imageCellIdentifier, for: indexPath)
+        cell.textLabel?.text = images[(indexPath as NSIndexPath).row]
```
Comment (sunshinejr) [inline Snappy/ImagesViewController.swift]: is the cast to `NSIndexPath` needed here? genuine question, not sure

File: Snappy/PhotoViewController.swift [hunk 3]
```diff
@@ -57,34 +57,34 @@ class PhotoViewController: UIViewController {
...
         CameraManager.switchCamera { [unowned self] sessionPreview in
             self.insertCameraPreview(sessionPreview)
         }
     }
     
-    @IBAction func flashButtonAction(sender: AnyObject) {
+    @IBAction func flashButtonAction(_ sender: AnyObject) {
         currentFlashMode = !currentFlashMode
-        flashButton.setImage(UIImage(named: currentFlashMode ? "FlashButtonImage" : "FlashOffButtonImage"), forState: .Normal)
+        flashButton.setImage(UIImage(named: currentFlashMode ? "FlashButtonImage" : "FlashOffButtonImage"), for: UIControlState())
```
Comment (sunshinejr) [inline Snappy/PhotoViewController.swift]: `.normal` instead of `UIConstrolState()`

File: Snappy/SnappyAPI.swift [hunk 4]
```diff
@@ -65,103 +65,86 @@ struct SnapchatAPIConstants {
...
-                switch result {
-                case .Success(let upload, _, _):
-                    upload.responseData { response in
-                        switch response.result {
-                        case .Success(let data):
-                            print("Success: \(String(data: data, encoding: NSUTF8StringEncoding))")
-                        case .Failure(let error):
-                            print("Error: \(error)")
-                        }
+       Alamofire.upload(multipartFormData: { (multipartData) in
```
Comment (sunshinejr) [inline Snappy/SnappyAPI.swift]: this closure with `multipartData` could be extracted before `Alamofire.upload`, to make it more readable (yeah, this on the left is not readable either :D)

File: SnappyAPITests/SnappyAPITests.swift [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    func testGetAllImages() {
+        let expectationGetAllImages = self.expectation(description: "getting all images")
+        
+        SnapchatAPI.getImages { (response) in
+            switch response {
+            case .success(let response):
+                let responseDict = response as? [String: Any]
+                let imageArray = responseDict?["images"] as? [[String: Any]]
+                XCTAssertTrue(imageArray!.count > 0)
+                expectationGetAllImages.fulfill()
```
Comment (sunshinejr) [inline SnappyAPITests/SnappyAPITests.swift]: hmm, maybe move it after the switch? in failure case it doesn't get fulfilled and waits for response

**Review by sunshinejr — CHANGES_REQUESTED** [review]
👌 just a few minor comments

**Comment by PChmiel** [comment]
@sunshinejr thx for cr, code updated

File: Snappy/SnappyAPI.swift [hunk 6]
```diff
@@ -95,23 +109,21 @@ struct SnapchatAPI {
...
+                   responseJSONClosure(response)
                 }
             case .failure(_):
                 completion(SnapchatAPIConstants.Error.alamofireEncodingError)
             }
         }
+        
+        // Using Alamofire we will upload the data on a server and return
+        // response with completion block
+        Alamofire.upload(multipartFormData: { (multipartData) in
```
Comment (sunshinejr) [inline Snappy/SnappyAPI.swift]: hmm, isn't it working with just:

```
Alamofire.upload(mulitpartFormData: multipartDataClosure, to: ...
```

?

**Comment by PChmiel** [comment]
@sunshinejr should be better now;)

**Review by sunshinejr — APPROVED** [review]
👏

**Pull request merged**